File: LinkedList.java

import java.util.Scanner;
public class LinkedList 
{ 
    static class Node 
    { 
        int data; 
        Node next; 

    } Node head; Node last;
    
    public static LinkedList insert(LinkedList list, int data) 
    { 
        Node new_node = new Node();
        
        new_node.data = data;
        new_node.next = null;
        
        if (list.head == null) 
        { 
            list.head = new_node;
            list.last = new_node;
        } 
        else 
        {  
            list.last.next = new_node;
            list.last = new_node; 
        } 
        return list; 
    }
    
    public static void printList(LinkedList list) 
    { 
        while (list.head != null) 
        { 
            System.out.print(list.head.data + " "); 
            list.head = list.head.next; 
        } 
    }
    
    public static void main(String[] args) 
    { 
        LinkedList list = new LinkedList();
		Scanner scan = new Scanner(System.in);
		int data;
		do
		{ 
			data = scan.nextInt();
			if(data > 0)
				list = insert(list,data);
		} while(data > 0);
		
		printList(list);
    } 
} 

File: LinkedList.cpp
#include<bits/stdc++.h> 
using namespace std;

class Node
{
	public:
		int data;
		Node * next;
} * head = NULL, *temp = NULL, * newnode = NULL;

void insert(Node **head, int data)
{
	newnode = new Node();
	newnode -> data = data;
	newnode -> next = NULL;
	
	if(*head == NULL)
	{
		*head = newnode;
		temp = newnode;
	}
	else
	{
		temp -> next = newnode;
		temp = newnode;
	}
}

void display(Node * head)
{
	while(head != NULL)
	{
		cout << head -> data << " ";
		head = head -> next;
	}
}

int main()
{
	int data;
	do
	{
		cin >> data;
		if(data > 0)
			insert(&head, data);
	}while(data > 0);
	
	display(head);
	
	
}












